lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -258,9 +263,15 @@ getTransaction = label "getTransaction" $ do
 
 putSignedTx :: (Tx, [TxWitness]) -> Put
 putSignedTx (tx@(Tx inputs outputs), witnesses) = withSizeHeader16be $ do
+    unless (length inputs < 256) $
+        fail "number of inputs must be lower than 256"
+    unless (length outputs < 256) $
+        fail "number of outputs must be lower than 256"
+    unless (length inputs == length witnesses) $
+        fail "number of witnesses must equal number of inputs"
     putWord8 2
-    putWord8 $ fromIntegral $ length inputs
-    putWord8 $ fromIntegral $ length outputs
+    putWord8 $ overflowSafeCoerce $ length inputs
```

Comment:
If we factored out the default `Typeable` error message from `overflowSafeCoerce`, we could remove the duplicated `< 256` checks.  Although since we're very untyped here, it might be nice to be very explicit about it.

Reply:
N/A. Checks are no longer duplicated.